### ISSUE TITLE
Ontario Health name data error and Apple Wallet name display error

### ISSUE THREAD

**Issue by scilib** [ongov/OpenVerify#18]
Sometime roughly after April 2020 the Ontario COVID-19 Proof of Vaccination data developed an error:
for some people, it has name as Firstname Middlename MIDDLENAME Lastname.

That is, it has added a second middlename in uppercase after the second.

If I were James Tiberius Kirk, the vaccine certificate would now show James Tiberius TIBERIUS Kirk.

It should be easy to do a database search to find these errors.

--

When the Verify Ontario app scans the QR code, it shows Firstname Middlename MIDDLENAME Lastname,
so clearly the data error is reproduced within the QR.

--

Separately, Apple Wallet vaccination QR code display seems to be confused by middle names.

It looks like it shows name as Firstname Middlename (or in my case Firstname Middlename MIDDLENAME), rather than Firstname Lastname.

--

I thought this might just be me, but I checked a friend's phone and he has the same Middlename MIDDLENAME display issue in Apple Wallet.

I know this may be out of scope but this seems a reasonable place to report it.

**Comment by spncrd** [comment]
I've shared this bug with the Ministry of Health IT Team.

**Comment by spncrd** [comment]
@scilib if you download your proof of vaccination again, can you confirm that it is in fact still on the PDF generated and in the QR code?

**Comment by scilib** [comment]
> @scilib if you download your proof of vaccination again, can you confirm that it is in fact still on the PDF generated and in the QR code?

Yes, I just downloaded it again now (November 4, ~8:30pm) and the error is still on both the PDF and in the QR code.

**Comment by spncrd** [comment]
Thanks so much for sharing @scilib. I've passed this along as well as your suspicion it might be DB related. You might also want to check if any other correspondence you've had with the Ministry of Heath reflects this same error. 

Regardless, to properly work this issue up and get it rectified for you personally, please call  1-833-943-3900 - they're open 8pm to 8pm everyday with rarely any wait times (especially during the afternoon). If you do get a resolution, please share what you know here.

**Comment by scilib** [comment]
> Thanks so much for sharing @scilib. I've passed this along as well as your suspicion it might be DB related.
> 
> Regardless, to properly work this issue up and get it rectified for you personally, please call 1-833-943-3900 - they're open 8pm to 8pm everyday with rarely any wait times (especially during the afternoon). If you do get a resolution, please share what you know here.

I was able to resolve the name error.  You can call the Ontario Provincial Vaccine Contact Centre at the 1-833 number above but they will just redirect you to your local public health authority who are the ones who can correct a name error.  

I had to navigate multiple levels of telephone number presses to get to the right location for my public health authority (in the end I had to connect to the local COVID-19 vaccine booking line and then select the option for "you require a receipt"), but once I did they corrected the error in seconds and now both my QR code printout and Apple Wallet display are correct.

**Comment by scilib** [comment]
> I was able to resolve the name error. You can call the Ontario Provincial Vaccine Contact Centre at the 1-833 number above but they will just redirect you to your local public health authority who are the ones who can correct a name error.
> 
> I had to navigate multiple levels of telephone number presses to get to the right location for my public health authority (in the end I had to connect to the local COVID-19 vaccine booking line and then select the option for "you require a receipt"), but once I did they corrected the error in seconds and now both my QR code printout and Apple Wallet display are correct.

After getting this fixed by my local public health authority, I went to download the certificate again today, just under a month since it was fixed, and it's incorrect again.  Clearly some Ontario provincial database is out there uncorrecting corrections made by local public health.  My name on the certificate is once again the equivalent of James Tiberius TIBERIUS Kirk.

**Comment by jfyu** [comment]
Quite surprised that out of all places where I have searched this issue, it turned up in Github...

Anyway I want to chime in my 2 cents: I was able to get my name fixed by Unity Health where I got my first and second dose. I just (literally an hour ago) got my booster from Shoppers and from checking my vaccine passport, the same issue happened again. Clearly whenever the vaccine QR code is updated, the lookup table from OHIP card to name is messed up somewhere and it is still not fixed. 

Given that there's a not-negligible possibility that we might need a 4th dose, it's probably advisable for this to be fixed so those of us affected don't have to waste public resources fixing trivial things like this after every booster....

**Comment by scilib** [comment]
> Quite surprised that out of all places where I have searched this issue, it turned up in Github...
> 
> Anyway I want to chime in my 2 cents: I was able to get my name fixed by Unity Health where I got my first and second dose. I just (literally an hour ago) got my booster from Shoppers and from checking my vaccine passport, the same issue happened again. Clearly whenever the vaccine QR code is updated, the lookup table from OHIP card to name is messed up somewhere and it is still not fixed.
> 
> Given that there's a not-negligible possibility that we might need a 4th dose, it's probably advisable for this to be fixed so those of us affected don't have to waste public resources fixing trivial things like this after every booster....

Yes, it appears to happen after every update to the QR code (after every new dose); I can verify it happens again after the 4th dose.

**Comment by jfyu** [comment]
For the next person who find this issue -- I just contacted my local PHU, the confirmation is that unless you got your vaccine from a PHU run clinic, you have to contact the place where you got the vaccine to fix the name issue. 

- if you got it from a hospital clinic, you need to contact that clinic
- If you got it from a pharmacy, contact that pharmacy
